lib/muontrap/port.ex

```diff
@@ -112,4 +121,12 @@ defmodule MuonTrap.Port do
 
   defp encode_acks_helper(full_acks, partial_acks),
     do: [:binary.copy(<<255>>, full_acks), partial_acks - 1]
+
+  defp maybe_start_timer(timeout) when is_integer(timeout),
+    do: Process.send_after(self(), :timeout, timeout)
```

Comment:
I'm worried about `:timeout` getting sent to the calling process for unrelated reasons since it's a pretty common atom to send. It might also be in the process mailbox already.

Perhaps using a ref like this:

```suggestion
  defp maybe_start_timer(timeout) when is_integer(timeout) do
    timeout_message = {:timeout, make_ref()}
    ref = Process.send_after(self(), timeout_message, timeout)
    {ref, timeout_message}
  end
```

Reply:
Good call. :+1: